api/machinemanager/machinemanager.go

```diff
@@ -49,6 +49,48 @@ func (client *Client) ForceDestroyMachines(machines ...string) ([]params.Destroy
 	return client.destroyMachines("ForceDestroyMachine", machines)
 }
 
+// DestroyMachinesWithParams removes the given set of machines, the semantics of which
+// is determined by the force and keep parameters.
+// TODO(wallyworld) - for Juju 3.0, this should be the preferred api to use.
+func (client *Client) DestroyMachinesWithParams(force, keep bool, machines ...string) ([]params.DestroyMachineResult, error) {
+	if !keep {
+		if force {
+			return client.ForceDestroyMachines(machines...)
+		}
+		return client.DestroyMachines(machines...)
+	}
+	args := params.DestroyMachinesParams{
+		Force:       force,
+		Keep:        keep,
+		MachineTags: make([]string, 0, len(machines)),
+	}
+	allResults := make([]params.DestroyMachineResult, len(machines))
```

Comment:
Was being consistent with existing code in this facade. The original thinking would have been to fail early if possible, avoiding an unnecessary api call if we know up front the tags are invalid. In fact, these sorts of checks are done in many other facades also. It's not uncommon to duplicate checks like this in different layers. We'll often also perform repeated checks in state and apiserver layers for example. Or CLI and api etc.